pennylane/fourier/simple_spectrum.py

```diff
@@ -246,12 +183,7 @@ def circuit(x):
     def wrapper(*args, **kwargs):
         qnode.construct(args, kwargs)
         tape = qnode.qtape
-
-        if encoding_gates is None:
-            freqs = {}
-        else:
-            freqs = {input_id: [] for input_id in encoding_gates}
-
+        freqs = {}
```

Comment:
Requested `id`s are not added at this point but in the end, if they were not picked up during the computation anyways.
This allows us to treat requested `id`s that are not even in the circuit differently than those that have a constant Fourier series.

Reply:
Good idea!